ContactDetailsApi/V2/Factories/EntityFactory.cs

```diff
@@ -100,26 +103,28 @@ private static string FormatSingleLineAddress(AddressExtended addressExtended)
         public static List<ContactDetails> ToDomain(this IEnumerable<ContactDetailsEntity> databaseEntity)
         {
             return databaseEntity
-                .Select(p => p.ToDomain())
-                .OrderBy(x => x.CreatedBy.CreatedAt)
+                .Select(p => p?.ToDomain())
+                .OrderBy(x => x?.CreatedBy?.CreatedAt)
                 .ToList();
         }
 
         // ServiceSoft Endpoint Factories
 
         public static PersonContactDetails ToUprnContact(this ContactDetails databaseEntity)
         {
+            if (databaseEntity == null) return null;
             return new PersonContactDetails
             {
                 Id = databaseEntity.Id,
                 ContactType = databaseEntity.ContactInformation.ContactType,
-                SubType = databaseEntity.ContactInformation.SubType,
+                SubType = databaseEntity.ContactInformation?.SubType == null ? null : databaseEntity.ContactInformation?.SubType,
                 Value = databaseEntity.ContactInformation.Value
             };
         }
 
         public static List<PersonContactDetails> ToContactByUprnPersonContacts(this IEnumerable<ContactDetails> databaseEntity)
         {
+            if (databaseEntity == null || databaseEntity.Count() == 0) return null;
```

Comment:
The `return null`

Are we sure about returning `null` collections? With REST the convention is: when no results are found empty collection is returned `[ ]`